### DISCUSSION TITLE
Possible issue in UCD composition table

### DISCUSSION THREAD

**Discussion by LaurenzV** [harfbuzz/harfbuzz#4795] [Q&A] [answered]
I'm currently trying to investigate a discrepancy between rustybuzz and harfbuzz and have identified the underlying cause, but I'm not sure who's right/wrong.

Given the[ following font](https://github.com/adobe-fonts/adobe-blank/blob/master/AdobeBlank.ttf) I'm running the following command with harfbuzz:
```
hb-shape AdobeBlank.ttf --unicodes U+0F42,U+0FB7
[cid01859=0+0|cid01976=0+0]
```

while in rustybuzz I get:
```
shape AdobeBlank.ttf --unicodes U+0F42,U+0FB7
cid01860=0+0
```

After some debugging, it seems to boil down that in the composition in harfbuzz, we have the following line:
`HB_CODEPOINT_ENCODE3 (0x0F42u, 0x0FB7u, 0x0000u)`. Since the third entry is 0, harfbuzz will return false in the `compose` function, while rustybuzz does find `\u{0F43}`. At least on [this page](https://www.compart.com/en/unicode/U+0F43), it seems like this character is indeed a composition of the two other characters. I'm not 100% sure though, is this a bug in harfbuzz or in rustybuzz?

**Comment by jfkthame** [comment, marked as answer]
The composition `U+0F42, U+0FB7` -> `U+0F43` (among others) is explicitly blocked in Unicode's NFC form; see https://www.unicode.org/reports/tr15/#Primary_Exclusion_List_Table and the data file https://www.unicode.org/Public/UCD/latest/ucd/CompositionExclusions.txt.

**Reply by LaurenzV** [reply]
Got it, guess I'll have to find the mistake in rustybuzz then, thanks!
